internal/cli/capact/build.go

```diff
@@ -0,0 +1,162 @@
+package capact
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/pkg/errors"
+)
+
+type image struct {
+	Dir    string
+	Target string
+
+	ExtraBuildArgs  []string
+	DisableBuildKit bool
+}
+
+var images = map[string]image{
+	"gateway": {
+		Dir:    ".",
+		Target: "generic",
+	},
+	"k8s-engine": {
+		Dir:    ".",
+		Target: "generic",
+	},
+	"hub-js": {
+		Dir: "hub-js",
+
+		DisableBuildKit: true,
+	},
+	"argo-runner": {
+		Dir:    ".",
+		Target: "generic",
+	},
+	"helm-runner": {
+		Dir:    ".",
+		Target: "generic",
+	},
+	"cloudsql-runner": {
+		Dir:    ".",
+		Target: "generic",
+	},
+	"terraform-runner": {
+		Dir:    ".",
+		Target: "terraform-runner",
+	},
+	"populator": {
+		Dir:    ".",
+		Target: "generic-alpine",
+	},
+	"e2e": {
+		Dir:    ".",
+		Target: "e2e",
+		ExtraBuildArgs: []string{
+			"BUILD_CMD=go test -v -c",
+			"SOURCE_PATH=./test/e2e/*_test.go",
+		},
+	},
+}
+
+func buildImage(w io.Writer, imgName string, img image, repository, version string) (string, error) {
+	// docker build --build-arg COMPONENT=$(APP) --target generic -t $(DOCKER_REPOSITORY)/$(APP):$(DOCKER_TAG)
+	imageTag := fmt.Sprintf("%s/%s:%s", repository, imgName, version)
+
+	// #nosec G204
+	cmd := exec.Command("docker",
+		"build",
+		"--build-arg", fmt.Sprintf("COMPONENT=%s", imgName),
+		"-t", imageTag, ".")
+
+	if img.Target != "" {
+		cmd.Args = append(cmd.Args, "--target", img.Target)
+	}
+
+	cmd.Env = os.Environ()
+
+	if !img.DisableBuildKit {
+		cmd.Env = append(cmd.Env,
+			// enable the BuildKit builder in the Docker CLI.
+			"DOCKER_BUILDKIT=1")
+	}
+
+	if len(img.ExtraBuildArgs) != 0 {
+		for _, arg := range img.ExtraBuildArgs {
+			cmd.Args = append(cmd.Args, "--build-arg", arg)
+		}
+	}
+	cmd.Dir = img.Dir
+	cmd.Stdout = w
+	cmd.Stderr = w
+	err := cmd.Run()
+	if err != nil {
+		return "", err
+	}
+	return imageTag, nil
+}
+
+// DeleteImages deletes passed image
+func DeleteImages(images []string) error {
```

Comment:
In bash scripts, we were deleting images after uploading them to kind. I don't have strong opinion if we should do it or not but it would prefer to delete not used code or use it.

Reply:
Will remove in the follow-up pr